### PR TITLE
Project api for electron application

### PR DESCRIPTION
**Pull request by reyery**
This PR add additional functions to the Flask server to enable the use of the upcoming electron version of the CEA.

The additions mainly include endpoints to create and manage project settings.

The pre-release version of the electron app can be downloaded here for testing. 
https://we.tl/t-5H3yJKhY08

### CODE REVIEW

**Comment by daren-thomas** [comment]
@reyery what exactly am I supposed to be testing here?

**Comment by reyery** [comment]
@daren-thomas mainly project management i.e. creating new project/scenario (especially the new scenario form), editing/deleting scenarios.

**Comment by daren-thomas** [comment]
I'm able to create new projects and scenarios. I think it's good as-is.

**Comment by daren-thomas** [comment]
Regarding the electron interface: if we add the link to the manual and issues (the documentation part), then we'd be good to go, no?

**Comment by daren-thomas** [comment]
I also tested with the browser dashboard. That still works. (it only has to work until we have an electron installer for windows, after that we can stop supporting it i guess)

File: cea/interfaces/dashboard/api/project.py [hunk 1]
```diff
@@ -20,18 +28,30 @@
...
+        if not os.path.exists(config.scenario):
+            config.scenario_name = ''
+            config.save()
+        return {'name': name, 'path': config.project, 'scenario': config.scenario_name,
+                'scenarios': config.get_parameter('general:scenario-name')._choices}
 
     @api.doc(body=PROJECT_PATH_MODEL, responses={200: 'Success', 400: 'Invalid Path given'})
-    def post(self):
-        config = cea.config.Configuration()
+    def put(self):
```
Comment (daren-thomas) [inline cea/interfaces/dashboard/api/project.py]: are you sure this isn't the wrong way round? `PUT` should be used to create a new one, `POST` to update an existing project?

Comment (reyery) [reply cea/interfaces/dashboard/api/project.py]: yeah, `POST` is to **create** and `PUT` is to **update/replace**

Comment (daren-thomas) [reply cea/interfaces/dashboard/api/project.py]: @reyery you're right! i had this wrong in my head all these years! thank you :)

**Pull request merged**